calc_with_colorize.rb

```diff
@@ -0,0 +1,117 @@
+# maps strings to correct function type
+require "colorize"
+
+def calculate(operation, first, second)
+  if operation == "add" || operation == "+" || operation == "addition"
+    add(first, second)
+  elsif operation == "subtract" || operation == "-" || operation == "subtraction"
+    subtract(first, second)
+  elsif operation == "multiply" || operation == "*" || operation == "multiplication"
+    multiply(first, second)
+  elsif operation == "divide" || operation == "/" || operation == "division"
+    divide(first, second)
+  elsif operation == "modulo" || operation == "%" || operation == "mod"
+    modulo(first, second)
+  elsif operation == "exponent" || operation == "^" || operation == "exp"
+    exponent(first, second)
+  end
+end
+
+# action of each type of math operation
+
+def add(first, second)
+  first + second
+end
+
+def subtract(first, second)
+  first - second
+end
+
+def multiply(first, second)
+  first * second
+end
+
+def divide(first, second)
+  first / second
+end
+
+def modulo(first, second)
+  first % second
+end
+
+def exponent(first, second)
+  first ** second
+end
+
+# map string to correct title of operation
+def op_name(operation)
+  if operation == "add" || operation == "+" || operation == "addition"
+    return "addition"
+  elsif operation == "subtract" || operation == "-" || operation == "subtraction"
+    return "subtraction"
+  elsif operation == "multiply" || operation == "*" || operation == "multiplication"
+    return "multiplication"
+  elsif operation == "divide" || operation =="/" || operation == "division"
+    return "division"
+  elsif operation == "modulo" || operation =="%" || operation == "mod"
+    return "modulo"
+  elsif operation == "exponent" || operation =="^" || operation == "exp"
+    return "exponent"
+  end
+end
+
+# map string to correct symbol of operation
+def op_sym(operation)
+  if operation == "add" || operation == "+" || operation == "addition"
+    return "+"
+  elsif operation == "subtract" || operation == "-" || operation == "subtraction"
+    return "-"
+  elsif operation == "multiply" || operation == "*" || operation == "multiplication"
+    return "*"
+  elsif operation == "divide" || operation =="/" || operation == "division"
+    return "/"
+  elsif operation == "modulo" || operation =="%" || operation == "mod"
+    return "%"
+  elsif operation == "exponent" || operation =="^" || operation == "exp"
+    return "^"
+  end
+end
```

Comment:
Consider reducing duplication (DRY - don't repeat yourself) by creating variables that you can re-use in each if/elsif block. It could look something like:

```ruby
if operation == "add" || operation == "+" || operation == "addition"
  op_name   = "add"
  op_symbol = "+"
elsif ...
```